RealmTasks Apple/RealmTasks macOS/ContainerViewController.swift

```diff
@@ -0,0 +1,168 @@
+        let listViewController = ListViewController(list: list)
+
+        addChildViewController(listViewController)
+        view.addSubview(listViewController.view)
+
+        if let currentListViewController = currentListViewController {
+            constrain(listViewController.view, currentListViewController.view, replace: constraintGroup) { newView, oldView in
+                oldView.edges == oldView.superview!.edges
+
+                if list is CellPresentable {
+                    newView.top == newView.superview!.bottom
+                } else {
+                    newView.bottom == newView.superview!.top
+                }
+
+                newView.left == newView.superview!.left
+                newView.size == newView.superview!.size
+            }
+
+            view.layoutSubtreeIfNeeded()
+
+            constrain(listViewController.view, currentListViewController.view, replace: constraintGroup) { newView, oldView in
+                newView.edges == newView.superview!.edges
+
+                if list is CellPresentable {
+                    oldView.bottom == oldView.superview!.top
+                } else {
+                    oldView.top == oldView.superview!.bottom
+                }
+
+                oldView.left == oldView.superview!.left
+                oldView.size == oldView.superview!.size
+            }
+
+            listViewController.view.alphaValue = 0
+
+            NSView.animate(duration: 0.3, animations: {
+                currentListViewController.view.alphaValue = 0
+                listViewController.view.alphaValue = 1
+
+                self.view.layoutSubtreeIfNeeded()
+            }) {
+                currentListViewController.removeFromParentViewController()
+                currentListViewController.view.removeFromSuperview()
+            }
+        } else {
+            constrain(listViewController.view, replace: constraintGroup) { view in
+                view.edges == view.superview!.edges
+            }
+        }
+
+        currentListViewController = listViewController
+
+        updateToolbarForList(list)
+
+        notificationToken?.stop()
+        notificationToken = list.realm?.addNotificationBlock { [unowned self] _, _ in
+            // Show all lists if list is deleted on other device
+            if (list as Object).invalidated {
+                self.showAllLists(nil)
+            }
+        }
+    }
+
+    private func updateToolbarForList<ListType: ListPresentable where ListType: Object>(list: ListType) {
+        guard let toolbar = view.window?.toolbar else {
+            return
+        }
+
+        if let titleLabel = toolbar.itemWithIdentifier("TitleLabel")?.view as? NSTextField {
+            titleLabel.stringValue = (list as? CellPresentable)?.text ?? "Lists"
+        }
+
+        if list is CellPresentable {
+            if !toolbar.hasItemWithIdentifier("ShowAllListsButton") {
```

Comment:
`"ShowAllListsButton"` is hardcoded 3 times here, could extract it into a constant just above.